internal/logger/target/http/http.go

```diff
@@ -374,30 +368,31 @@ func (h *Target) startQueueProcessor(ctx context.Context, mainWorker bool) {
 					h.failedMessages.Add(1)
 					continue
 				}
-				count++
 			} else {
 				entries = append(entries, entry)
-				count++
 			}
+			count++
 		}
 
-		if count != h.batchSize {
-			if len(h.logCh) > 0 || len(globalBuffer) > 0 || count == 0 {
-				continue
-			}
+		if len(h.logCh) > 0 || len(globalBuffer) > 0 || count == 0 {
```

Comment:
Suppose `count == h.batchSize` before the if block at line 359 (so the if block is not entered), then if line 377 evaluates to true, looks like current `entry` is lost due to being overwritten?

Reply:
good point, more than one entry can be lost if less than one second elapsed since last batch

Reply:
so, it should simply be `if count <= h.batchSize {`